### ISSUE TITLE
Dynamics pen pressure not working on Linux with XP-PEN drawing tablet

### ISSUE THREAD

**Issue by nmdonaldson** [aseprite/aseprite#3125]
## Issue

The dynamics settings became unable to detect my tablet's pen pressure after switching from Windows 10 to Linux. It works fine in other art programs like Krita, and in Windows before I switched. This is with the pentablet driver straight from XP-Pen on both OSes. Using the wacom driver as a last resort didn't help; neither did trying various versions of Proton. 

However, I discovered a workaround!

## Workaround

The name of my tablet's stylus as X.org sees it can be shown with `xinput --list`. In my case, it's `XP-PEN STYLUS Pen (0)` with `id=24`:

![image](https://user-images.githubusercontent.com/54925681/149060533-d20df752-d58a-47e1-acff-67dfccf8e10d.png)

With `xinput test 24` (since the ID of my pen is 24), it detects the pen and its pressure just fine:

![image](https://user-images.githubusercontent.com/54925681/149064573-35d24132-9e32-474e-afff-500dea0e0c3d.png)

I figured out that Aseprite's dynamics backend in  `/aseprite/laf/os/x11/xinput.h` looks for the substring "stylus" from xinput for the pen, but in my scenario, it doesn't match since `std::strstr()` is case sensitive.

![image](https://user-images.githubusercontent.com/54925681/149062277-620cd085-3586-45d8-9b25-3b83d4c276fb.png)

I changed it to look for "STYLUS" in the device name to match the cases properly, recompiled Aseprite's most recent beta from source, and found that it works now!

![image](https://user-images.githubusercontent.com/54925681/149062827-615a0b88-e2fd-425b-b684-76e7018359d5.png)
![image](https://user-images.githubusercontent.com/54925681/149062924-337ff6a5-970d-44e5-a9d6-e52f3e2fc2d1.png)

However, I would still prefer a more permanent solution than this for the 3 people out there with my OS/tablet combination.

### Aseprite and System version

* Aseprite versions: v1.3-beta7, v1.2.30 | both on steam (with/without Proton, I tried all)
* System: Arch Linux 5.15.13-arch1-1
* Tablet: XP-Pen Artist 22E Pro

**Comment by dacap** [comment]
Hi @nmdonaldson, I was going to release a new version just today (I've already the packages compiled/ready), but this fix is so simple that I might be able to sneak it around in v1.2.31 and v1.3-beta8.

**Comment by Anuken** [comment]
I have the same issue, except my tablet device (Huion Kamvas Pro 16) doesn't even have the word "stylus" in it:

![image](https://user-images.githubusercontent.com/10100000/149871910-41854089-accc-4c00-b77b-81b1308b093e.png)

This may be related to the fact that I'm using a strange driver. Adding in ` || std::strstr(devName.c_str(), " pen")` fixes the problem, although I'm not sure if checking for " pen" could break other devices. Thoughts?

**Comment by bferguson3** [comment]
Bump because I have the same issue. xinput --list with XP-PEN G960S:
```
⎡ Virtual core pointer                    	id=2	[master pointer  (3)]
⎜   ↳ Virtual core XTEST pointer              	id=4	[slave  pointer  (2)]
⎜   ↳ vc4                                     	id=15	[slave  pointer  (2)]
⎜   ↳ UGTABLET 9 inch PenTablet Mouse         	id=9	[slave  pointer  (2)]
⎜   ↳ PixArt USB Optical Mouse                	id=8	[slave  pointer  (2)]
⎜   ↳ vc4                                     	id=14	[slave  pointer  (2)]
⎜   ↳ UGTABLET 9 inch PenTablet Pen (0)       	id=18	[slave  pointer  (2)]
⎜   ↳ UGTABLET 9 inch PenTablet Eraser (0)    	id=19	[slave  pointer  (2)]
⎣ Virtual core keyboard                   	id=3	[master keyboard (2)]
    ↳ Virtual core XTEST keyboard             	id=5	[slave  keyboard (3)]
    ↳   Raspberry Pi Internal Keyboard        	id=13	[slave  keyboard (3)]
    ↳ vc4                                     	id=17	[slave  keyboard (3)]
    ↳   Raspberry Pi Internal Keyboard        	id=12	[slave  keyboard (3)]
    ↳ UGTABLET 9 inch PenTablet Keyboard      	id=10	[slave  keyboard (3)]
    ↳ UGTABLET 9 inch PenTablet               	id=11	[slave  keyboard (3)]
    ↳   RPI Wired Keyboard 7                  	id=7	[slave  keyboard (3)]
    ↳   RPI Wired Keyboard 7                  	id=6	[slave  keyboard (3)]
    ↳ vc4                                     	id=16	[slave  keyboard (3)]
```
In my case the correct IDs are 18 and 19

**Comment by dacap** [comment]
@Anuken definitely `" pen"` cannot be used (as it would match other devices as @bferguson3 show, it could match several `" pentablet"` devices). I will include a complete match for `"tablet monitor pen"`.

**Comment by Eve1374** [comment]
Hello, I have a problem with my pen too, it is recognised as a keyboard. I have a surface go 3 with debian bookworm (`uname -a ` > `Linux eve-surface 6.1.0-7-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.20-2 (2023-04-08) x86_64 GNU/Linux` ) and a HP active pen G3. Here is xinput --list output (I'm not able to quote it properly...). Is it related to the fact that it is not named stylus something ?

xinput --list
⎡ Virtual core pointer                               id=2    [master pointer  (3)]
⎜   ↳ Virtual core XTEST pointer              id=4    [slave  pointer  (2)]
⎜   ↳ ELAN9038:00 04F3:2A1C                id=8    [slave  pointer  (2)]
⎣ Virtual core keyboard                           id=3    [master keyboard (2)]
    ↳ Virtual core XTEST keyboard           id=5    [slave  keyboard (3)]
    ↳ Video Bus                                        id=6    [slave  keyboard (3)]
    ↳ Power Button                                  id=7    [slave  keyboard (3)]
    ↳ ELAN9038:00 04F3:2A1C Stylus      id=9    [slave  keyboard (3)]
    ↳ Intel HID events                              id=10   [slave  keyboard (3)]
    ↳ Intel HID 5 button array                  id=11   [slave  keyboard (3)]
    **↳ HP Active Pen G3 Keyboard         id=12   [slave  keyboard (3)]**

**Comment by dacap** [comment]
Hi @Eve1374 check the solution in https://www.aseprite.org/docs/tablet#linux/x11, probably adding 
```
[general]
x11_stylus_id = Active Pen
```

**Comment by Eve1374** [comment]
Hi, thank you for your reply.
I'm afraid I missed something, I don't understand what is aseprite / what it has to do with pen support ?

Anyway when I run their command, here is what I have : 

sudo xdg-open ~/.config/aseprite
[sudo] password for eve: 
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
[kf.service.services](https://kf.service.services/): KApplicationTrader: mimeType "x-scheme-handler/file" not found
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
/usr/bin/xdg-open: 491: test: Illegal number: 
/usr/bin/xdg-open: 492: test: Illegal number: 
/usr/bin/xdg-open: 493: test: Illegal number:

**Comment by dacap** [comment]
Anyway @Eve1374 as you said that you don't know what is "aseprite," I think you might be in the wrong forum. This issue tracker is for Aseprite, not for generic Linux problems with stylus/pens.

**Comment by Eve1374** [comment]
Hello,

> Anyway @Eve1374 as you said that you don't know what is "aseprite," I think you might be in the wrong forum. This issue tracker is for Aseprite, not for generic Linux problems with stylus/pens.

Indeed, you are totally right and I apologize for having posted my message here. I arrived here directly from search engine so I didn't went through the categories and noticed it was for aseprite after posting (+ the issue of the pen not recognised by xinput is very similar to mine). Perhaps I should install aseprite to make my pen work ;)

Also I tried both root and non root to open the aseprite config file but no surprise, none works.

In the meantime I'll dig for solutions on other threads.

Thank you and sorry again for misplaced post.
